components/scream/src/physics/spa/tests/spa_one_to_one_remap_test.cpp

```diff
@@ -0,0 +1,149 @@
+  Int nlwbands = 3;
+
+  // Break the test set of columns into local degrees of freedom per mpi rank
+  auto comm_size = spa_comm.size();
+  auto comm_rank = spa_comm.rank();
+
+  int my_ncols = ncols/comm_size + (comm_rank < ncols%comm_size ? 1 : 0);
+  view_1d<gid_type> dofs_gids("",my_ncols);
+  gid_type min_dof = 0;  // We will set up the dof's to start from 0
+  Kokkos::parallel_for("", my_ncols, KOKKOS_LAMBDA(const int& ii) {
+    dofs_gids(ii) = min_dof + static_cast<gid_type>(comm_rank + ii*comm_size);
+  });
+  // Make sure that the total set of columns has been completely broken up.
+  Int test_total_ncols = 0;
+  spa_comm.all_reduce(&my_ncols,&test_total_ncols,1,MPI_SUM);
+  REQUIRE(test_total_ncols == ncols);
+
+  // Set up the set of SPA structures needed to run the test
+  SPAFunc::SPAHorizInterp spa_horiz_interp;
+  spa_horiz_interp.m_comm = spa_comm;
+  SPAFunc::set_remap_weights_one_to_one(ncols,min_dof,dofs_gids,spa_horiz_interp);
+  SPAFunc::SPAData spa_data(dofs_gids.size(), nlevs, nswbands, nlwbands);
+
+  // Check that the horizontal interpolation data is in fact a 1-1 mapping
+  Kokkos::parallel_for("", spa_horiz_interp.length, KOKKOS_LAMBDA(const int& ii) {
+    EKAT_KERNEL_ASSERT(spa_horiz_interp.weights(ii)==1.0);
+    EKAT_KERNEL_ASSERT(spa_horiz_interp.source_grid_loc(ii)==dofs_gids(ii)-min_dof);
+    EKAT_KERNEL_ASSERT(spa_horiz_interp.target_grid_loc(ii)==ii);
+  });
+
+  // Verify that the interpolated values match the algorithm for the data and the weights.
+  //       weights(i) = 1.0 
+  //       FOR t=1,2,3; i=0,1,2; b=1,2 or 1,2,3 and k=0,1,2,3
+  //       p(t,i) = (t+1) * (i+1)*100
+  //       ccn3(t,i,k) = (i+1)*100 + t*10 + k
+  //       aer_g_sw(t,i,b,k) = t
+  //       aer_ssa_sw(t,i,b,k) = i
+  //       aer_tau_sw(t,i,b,k) = b
+  //       aer_tau_lw(t,i,b,k) = k
+  auto ps_h         = Kokkos::create_mirror_view(spa_data.PS);
+  auto ccn3_h       = Kokkos::create_mirror_view(spa_data.CCN3);
+  auto aer_g_sw_h   = Kokkos::create_mirror_view(spa_data.AER_G_SW);
+  auto aer_ssa_sw_h = Kokkos::create_mirror_view(spa_data.AER_SSA_SW);
+  auto aer_tau_sw_h = Kokkos::create_mirror_view(spa_data.AER_TAU_SW);
+  auto aer_tau_lw_h = Kokkos::create_mirror_view(spa_data.AER_TAU_LW);
+  auto dofs_gids_h = Kokkos::create_mirror_view(dofs_gids);
+  Kokkos::deep_copy(dofs_gids_h,dofs_gids);
+  for (int time_index = 0;time_index<max_time; time_index++) {
+    SPAFunc::update_spa_data_from_file(spa_data_file, time_index+1, nswbands, nlwbands,
+                                       spa_horiz_interp, spa_data);
+    Kokkos::deep_copy(ps_h,spa_data.PS);
+    Kokkos::deep_copy(ccn3_h,spa_data.CCN3);
+    Kokkos::deep_copy(aer_g_sw_h,spa_data.AER_G_SW);
+    Kokkos::deep_copy(aer_ssa_sw_h,spa_data.AER_SSA_SW);
+    Kokkos::deep_copy(aer_tau_sw_h,spa_data.AER_TAU_SW);
+    Kokkos::deep_copy(aer_tau_lw_h,spa_data.AER_TAU_LW);
+    for (int dof_i=0;dof_i<dofs_gids_h.size();dof_i++) {
+      gid_type glob_i = dofs_gids_h(dof_i);
```

Comment:
Using `glob_i` to access views might cause OOB errors if the gids do not start from 0... You should prob compute `glob_i` subtracting the min gid.

Reply:
`glob_i` is just used as an argument to the following functions, which calculate what the real value should be based on the global column id.

Reply:
But it's also used to access some host views, like `ps_h`, `ccn3_h`,...

Reply:
Ah you're right, I confused `dof_i` with `glob_i` while reading.